tools/gfx/render-d3d12.cpp

```diff
@@ -1388,32 +1339,95 @@ Result D3D12Renderer::initialize(const Desc& desc, void* inWindowHandle)
         }
     }
 
-    if (!adapter)
+    if (m_dxDebug && (deviceCheckFlags & DeviceCheckFlag::UseDebug))
     {
-        // Couldn't find an adapter
+        m_dxDebug->EnableDebugLayer();
+    }
+
+    outDxgiFactory = dxgiFactory;
+    outAdapter = adapter;
+    return SLANG_OK;
+}
+
+Result D3D12Renderer::initialize(const Desc& desc, void* inWindowHandle)
+{
+    m_hwnd = (HWND)inWindowHandle;
+    // Rather than statically link against D3D, we load it dynamically.
+
+    HMODULE d3dModule = LoadLibraryA("d3d12.dll");
+    if (!d3dModule)
+    {
+        fprintf(stderr, "error: failed load 'd3d12.dll'\n");
         return SLANG_FAIL;
     }
 
-    // set up debug layer
-#ifndef NDEBUG
+    HMODULE dxgiModule = LoadLibraryA("Dxgi.dll");
```

Comment:
On my machine at least, the actual file is `dxgi.dll` without a capital `D`. This won't matter in the 99% case because of case-insensitive file systems being standard on Windows, but it seems best to just match the capitalization we expect the file to have anyway.